services/getstream/followTopic.js

```diff
@@ -29,20 +32,66 @@ const followTopics = async (token, userIds) => {
 const followMainFeedTopic = async (token, userId, name) => {
   name = name.toLowerCase();
   const client = stream.connect(process.env.API_KEY, token, process.env.APP_ID);
-  const feed = client.feed(MAIN_FEED_FOLLOWING_NAME, client.userId, token);
+  const feed = client.feed(MAIN_FEED_FOLLOWING_NAME, userId, token);
   return feed.follow(TOPIC_FEED_NAME, name);
 };
 
 const unfollowMainFeedTopic = async (token, userId, name) => {
   name = name.toLowerCase();
   const client = stream.connect(process.env.API_KEY, token, process.env.APP_ID);
-  const feed = client.feed(MAIN_FEED_FOLLOWING_NAME, client.userId, token);
+  const feed = client.feed(MAIN_FEED_FOLLOWING_NAME, userId, token);
   return feed.unfollow(TOPIC_FEED_NAME, name);
 };
 
+const addTopicToChatTab = async (token, topicName, userId, isAnonymous) => {
+  try {
+    const topic = topicName.toLowerCase();
+    const text = isAnonymous
+      ? 'This topic has new anonymous followers'
+      : 'This topic has new followers';
+    const defaultImage = isAnonymous ? DEFAULT_TOPIC_PIC_PATH_ANON : DEFAULT_TOPIC_PIC_PATH_SIGN;
+
+    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    await client.connectUser({id: userId}, token);
+    const channel = await client.channel('topics', `topic_${topic}`, {
+      name: `#${topic}`,
+      members: [userId],
+      channel_type: 3,
+      channel_image: defaultImage,
+      channelImage: defaultImage,
+      image: defaultImage
+    });
+
+    console.log('prepare follow channel', topic);
+    await channel.create();
+    await channel.addMembers([userId]);
+    await channel.sendMessage({text}, {skip_push: true});
+    console.log('channel followed');
+  } catch (error) {
+    console.log('error add topic to chat tab', error);
+    throw new Error('error add topic to chat tab');
+  }
+};
+
+const removeTopicFromChatTab = async (token, topicName, userId) => {
+  try {
+    const name = topicName.toLowerCase();
+    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    await client.connectUser({id: userId}, token);
+    const channel = await client.channel('topics', `topic_${name}`);
+
+    await channel.removeMembers([userId]);
+  } catch (error) {
+    console.log('error remove topic to chat tab', error);
+    throw new Error('error remove topic to chat tab');
+  }
+};
+
 module.exports = {
   followTopic,
   followTopics,
   followMainFeedTopic,
-  unfollowMainFeedTopic
+  unfollowMainFeedTopic,
+  addTopicToChatTab,
+  removeTopicFromChatTab
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `addTopicToChatTab` and `removeTopicFromChatTab` functions have been added to the exports of the module. This is a good change as it allows these functions to be imported and used in other parts of the codebase.

```diff
+  module.exports = {
+    followTopic,
+    followTopics,
+    followMainFeedTopic,
+   unfollowMainFeedTopic,
+   addTopicToChatTab,
+   removeTopicFromChatTab
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->